app/common/services/data-export.service.js

```diff
@@ -0,0 +1,127 @@
+module.exports = DataExport;
+
+
+DataExport.$inject = ['$rootScope', 'ConfigEndpoint', 'ExportJobEndpoint', '$q', '_', '$window', '$timeout', 'Notify', '$location', '$interval'];
+function DataExport($rootScope, ConfigEndpoint, ExportJobEndpoint, $q, _, $window, $timeout, Notify, $location, $interval) {
+    function startExport(query) {
+        ExportJobEndpoint.save(query).$promise.then(function (job) {
+            loadingStatus(true, null, job.id);
+            startPolling(job);
+        }, function (err) {
+            loadingStatus(false, err);
+        });
+    }
+
+    function startPolling(job) {
```

Comment:
Is this supposed to continuously (recursively) run until it comes back as 'done' or 'err'? Not sure I understand the purpose of this function?

Reply:
Yep, its still not complete, but that is the idea, to ask the server 'is it done yet, is it done yet, is it done yet' until its done...